contracts/math/uints.ral

```diff
@@ -90,5 +90,74 @@ Abstract Contract U512Impl () {
                 lower: deltaL
             }
         }
+
+    pub fn bigAdd512(a: U512, b: U512) -> U512 {
+        let (r0, overflow) = overflowingAdd(a.lower, b.lower)
+        let r1 = wrappingAdd(wrappingAdd(a.higher, b.higher), overflow)
+        return U512 { higher: r1, lower: r0 }
+    }
+
+    pub fn bigMul(a: U256, b: U256, denominator: U256) -> U512 {
```

Comment:
We have a convention that the result of multiplication `a * b` is returned includes `a` denominator. However, this is just convention, and the result could be returned with the denominator `b`.

That's why IMO we should change the name `denominator` to `bDenominator`. Please make this change in the remaining places as well.